Enable `rosbag2_performance_benchmarking` package to be built by default

MichaelOrlov opened the pull request:
## Description

- This PR enables `rosbag2_performance_benchmarking` package to be built by default.

<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.*
-->

- Fixes #2037

### Is this user-facing behavior change?
No.

### Did you use Generative AI?
Yes, I used GitHub Copilot, Claude Sonnet 3.7 Thinking to help with writing comments and updates in the readme file.

### Additional Information
Note: the tests from the `rosbag2_performance_benchmarking` package are excluded from the build by default and should be explicitly enabled during the build time via ``--cmake-args -DENABLE_ROSBAG2_BENCHMARK_TESTS=1`

- This PR replaces https://github.com/ros2/rosbag2/pull/2073

MichaelOrlov committed:
Enable performance benchmarking to be built by default

- Note: the tests from the performance benchmarking package are excluded
 from the build by default and should be explicitly enabled during the
 build time via ``--cmake-args -DENABLE_ROSBAG2_BENCHMARK_TESTS=1`

Signed-off-by: Michael Orlov <morlovmr@gmail.com>

MichaelOrlov commented:
Pulls: ros2/rosbag2#2093
Gist: https://gist.githubusercontent.com/MichaelOrlov/d30a9499d685c574b0fd845e425a141e/raw/1e865c8607dd3a4d282dbe9b578cbaaffc903762/ros2.repos
BUILD args:  --packages-above-and-dependencies rosbag2_performance_benchmarking
TEST args:  --packages-above rosbag2_performance_benchmarking
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/16605
* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=24453)](http://ci.ros2.org/job/ci_linux/24453/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=18519)](http://ci.ros2.org/job/ci_linux-aarch64/18519/)
* Linux-rhel [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-rhel&build=3896)](http://ci.ros2.org/job/ci_linux-rhel/3896/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=25012)](http://ci.ros2.org/job/ci_windows/25012/)

christophebedard commented:
Why not go with #2073?

christophebedard commented on rosbag2_performance/rosbag2_performance_benchmarking/README.md:
```diff
@@ -45,12 +46,25 @@ In the case of the `benchmark_publishers` binary, a pool of threads is created t
+Rosbag2 recorder. However, these tests excluded from the build by default, so you need to enable
+them by setting the `ENABLE_ROSBAG2_BENCHMARK_TESTS` CMake flag to `ON` when building the package.
+To enable and run the rosbag2 benchmark tests with colcon, you need to:
+First build the package with the `ENABLE_ROSBAG2_BENCHMARK_TESTS` CMake option enabled:
```
```suggestion
The package contains a set of builtin tests that can be run to verify the performance of the
Rosbag2 recorder. However, these tests are excluded from the build by default, so you need to enable
them by setting the `ENABLE_ROSBAG2_BENCHMARK_TESTS` CMake flag to `ON` when building the package.
To enable and run the rosbag2 benchmark tests with colcon, you need to:
First build the package with the `ENABLE_ROSBAG2_BENCHMARK_TESTS` CMake option enabled:
```

MichaelOrlov replied:
Fixed in the https://github.com/ros2/rosbag2/pull/2093/commits/90c83512e1e902b6c825b9ffd89997fa992c7976 commit

christophebedard reviewed (approved):
Other than a simple fix in the README, this LGTM.

MichaelOrlov commented:
As regards. 
> Why not go with https://github.com/ros2/rosbag2/pull/2073?

Because it is targeted at the Humble and there has been no response from the author for almost two weeks. 
I want to move forwad and wrap it up.

MichaelOrlov committed:
Fix grammar issue in the README.md file according to the code review

Signed-off-by: Michael Orlov <morlovmr@gmail.com>

MichaelOrlov committed:
Fix for Windows build failure

Address error message:
 LINK : fatal error LNK1181: cannot open input file 'yaml-cpp.lib'

Signed-off-by: Michael Orlov <morlovmr@gmail.com>

MichaelOrlov commented:
Re-run Windows CI job after attempt to fix linker error about missing yaml-cpp library
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=25015)](http://ci.ros2.org/job/ci_windows/25015/)

MichaelOrlov commented:
The Windows CI job failed due to the inner infrastructure issue
```bash
Step 28/35 : RUN C:\TEMP\rticonnextdds-src\rti_connext_dds-7.3.0-pro-host-x64Win64.exe --mode unattended --unattendedmodeui minimalWithDialogs --prefix C:\connext
 ---> Running in eaa601dc3520
unknown parent image ID sha256:d06ffbd5a9e61400a8c2a5838b548c6c264f7de09b0f57e88526d44629729afc
Build step 'Execute Windows batch command' marked build as failure
```

Re-running one more time
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=25017)](http://ci.ros2.org/job/ci_windows/25017/)

MichaelOrlov commented:
https://github.com/Mergifyio backport kilted jazzy humble
